src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HPack/HeaderField.cs

```diff
@@ -30,5 +32,17 @@ public HeaderField(ReadOnlySpan<byte> name, ReadOnlySpan<byte> value)
         public int Length => GetLength(Name.Length, Value.Length);
 
         public static int GetLength(int nameLength, int valueLenth) => nameLength + valueLenth + RfcOverhead;
+
+        public override string ToString()
+        {
+            if (Name?.Length > 0)
+            {
+                return Encoding.ASCII.GetString(Name) + ": " + Encoding.ASCII.GetString(Value ?? Array.Empty<byte>());
```

Comment:
Can Value actually be null if Name is not?

I'm trying to understand the expected state here. I assume it's either Name=null and Value=null, which is the case when the entry is not in use; or, neither are null and Name is at least 1 byte long, when the entry is in use. Is that correct? Seems like the code here should reflect that.

Reply:
This is a struct, so default value these will be `null`. The `DynamicTable` buffer is an array of these where entries not in use will have null.

Reply:
Oh, I see what you mean. Yes, `Value` should not be null if `Name` isn't. I'll update.